internal/subscription/model.go

```diff
@@ -21,7 +22,7 @@ type Model struct {
 	ExtractedSubscriptionProjectID string
 	ExtractedTopicName             string
 	ExtractedSubscriptionName      string
-	Auth                           *Auth
+	Credentials                    credentials.ICredentials
```

Comment:
Should we have dependency on client facing model here for authentication.
Authentication of PushEndpoint and client may vary, probably maintain the Model defn within subscription itself

Reply:
The password encoding and decoding is handled with credentials pkg. Would need to duplicate that logic here if a new model is created.